src/Microsoft.OData.Core/UriParser/Resolver/UnqualifiedODataUriResolver.cs

```diff
@@ -46,28 +46,42 @@ public override IEnumerable<IEdmOperation> ResolveBoundOperations(IEdmModel mode
             {
                 return base.ResolveBoundOperations(model, identifier, bindingType);
             }
-
-            return FindAcrossModels<IEdmOperation>(model, identifier, this.EnableCaseInsensitive)
-                .Where(operation =>
-                    operation.IsBound
-                    && operation.Parameters.Any()
-                    && operation.HasEquivalentBindingType(bindingType));
+          
+            return GetBoundOperationsForModel(model, identifier, bindingType, this.EnableCaseInsensitive? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal);
         }
 
-        private static IEnumerable<T> FindAcrossModels<T>(IEdmModel model, String qualifiedName, bool caseInsensitive) where T : IEdmSchemaElement
+        private static IEnumerable<IEdmOperation> GetUnBoundOperationsForModel(IEdmModel model, string operationName, StringComparison strComparison)
         {
-            Func<IEdmModel, IEnumerable<T>> finder = (refModel) =>
-                refModel.SchemaElements.OfType<T>()
-                .Where(e => string.Equals(qualifiedName, e.Name, caseInsensitive ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal));
-
-            IEnumerable<T> results = finder(model);
+            foreach (IEdmSchemaElement schemaElement in model.SchemaElements)
+            {
+                if (schemaElement is IEdmOperation operation && !operation.IsBound && string.Equals(operationName, operation.Name, strComparison))
+                {
+                    yield return operation;
+                }
+            }
 
-            foreach (IEdmModel reference in model.ReferencedModels)
+            foreach(IEdmModel reference in model.ReferencedModels)
             {
-                results.Concat(finder(reference));
+                foreach (IEdmSchemaElement schemaElement in reference.SchemaElements)
+                {
+                    if (schemaElement is IEdmOperation operation && !operation.IsBound && string.Equals(operationName, operation.Name, strComparison))

```

Comment:
I'm assuming the two `if` statements are intended to be an identical predicate for all of the schema elements available. If that's the case, it might make more sense to combine them into one:

```
return model
  .SchemaElements
  .Concat(model.ReferencedModels.SelectMany(reference => reference.SchemaElements))
  .Where(schemaElement => schemaElement is IEdmOperation operation && !operation.IsBound && string.Equals(operationName, operation.Name, strComparison));
```

Reply:
Without knowing how they were optimized, I would be a bit concerned about the overhead of the concat and SelectMany.  General guidance as been to avoid LINQ in perf sensitive scenarios.